handler/request.go

```diff
@@ -24,13 +26,14 @@ func (wasmRequest) SetMethod(method string) {
 }
 
 // GetURI implements the same method as documented on api.Request.
-func (wasmRequest) GetURI() string {
-	return mem.GetString(imports.GetURI)
+func (wasmRequest) GetURI() url.URL {
+	u, _ := url.Parse(mem.GetString(imports.GetURI))
+	return *u
 }
 
 // SetURI implements the same method as documented on api.Request.
-func (wasmRequest) SetURI(uri string) {
-	ptr, size := mem.StringToPtr(uri)
+func (wasmRequest) SetURI(uri url.URL) {
```

Comment:
It is probably more convenient and less allocations if we accept `*url.URL` instead of `url.URL` as even if we don't keep the pointer but the value, less indirection and entire copies. WDYT @anuraaga ?

Reply:
Yeah let's use `*url.URL` for both, `url.Parse` also returns a pointer so it'll be rare to have a non-pointer in much code